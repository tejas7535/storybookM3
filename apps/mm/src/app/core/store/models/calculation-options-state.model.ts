import { PreflightData } from '@mm/core/services/preflght-data-parser/preflight-data.interface';

export interface CalculationOptionsState {
  options: PreflightData;
}
