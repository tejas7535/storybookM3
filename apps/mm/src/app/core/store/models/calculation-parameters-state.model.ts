export interface CalculationParametersState {
  parameters?: CalculationParameters;
}

export interface CalculationParameters {
  RSY_BEARING_TYPE: number;
  RSY_BEARING_SERIES: string;
  IDCO_DESIGNATION: string;
  IDMM_BEARING_SEAT: string;
  IDMM_MEASSURING_METHOD: string;
  IDMM_MOUNTING_METHOD: string;
  IDMM_HYDRAULIC_NUT_TYPE: string;
  IDMM_NUMBER_OF_PREVIOUS_MOUNTINGS: string;
  IDMM_CLEARANCE_REDUCTION_INPUT: string;
  IDMM_INNER_RING_EXPANSION: number;
  IDMM_RADIAL_CLEARANCE_REDUCTION: number;
  IDMM_INNER_SHAFT_DIAMETER: number;
  IDMM_SHAFT_MATERIAL: string;
  IDMM_MODULUS_OF_ELASTICITY: number;
  IDMM_POISSON_RATIO: number;
}
