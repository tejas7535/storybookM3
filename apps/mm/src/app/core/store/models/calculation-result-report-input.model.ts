export interface CalculationResultReportInput {
  hasNestedStructure: boolean;
  meaningfulRound?: boolean;
  title?: string;
  designation?: string;
  abbreviation?: string;
  unit?: string;
  value?: string;
  subItems?: CalculationResultReportInput[];
  titleID?: string;
}
