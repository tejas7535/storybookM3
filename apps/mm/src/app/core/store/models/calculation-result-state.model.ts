import { CalculationResultReportInput } from './calculation-result-report-input.model';

export interface CalculationResultState {
  isLoading: boolean;
  result?: CalculationResult;
  error?: string;
  versions?: { [key: string]: string };
}

export interface CalculationResult {
  inputs: CalculationResultReportInput[];
  startPositions: ResultItem[];
  endPositions: ResultItem[];
  mountingTools: MountingTools;
  mountingRecommendations: string[];
  radialClearance: ResultItemWithTitle[];
  clearanceClasses: ResultItemWithTitle[];
  reportMessages: ReportMessages;
}

export interface ResultItem {
  value: string;
  unit: string;
  abbreviation?: string;
  designation: string;
}

export interface ResultItemWithTitle {
  title: string;
  items: ResultItem[];
}

export interface PumpItem {
  field: string;
  value: string;
  isRecommended: boolean;
}

export interface MountingTools {
  additionalTools: ResultItem[];
  hydraulicNut: ResultItem[];
  pumps: {
    title: string;
    items: PumpItem[];
  };
  locknut: ResultItem[];
  sleeveConnectors: ResultItem[];
}

export interface ReportMessages {
  warnings: string[];
  errors: string[];
  notes: string[];
}

export type ResultType =
  | 'reportInputs'
  | 'startPosition'
  | 'endPosition'
  | 'mountingToolsAndUtilities'
  | 'mountingInstructions'
  | 'reportMessages';

export interface ResultTypeConfig {
  name: ResultType;
}
