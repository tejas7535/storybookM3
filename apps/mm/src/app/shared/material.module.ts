import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatExpansionModule } from '@angular/material/expansion';
import { MatIconModule } from '@angular/material/icon';
import { MatLegacyButtonModule as MatButtonModule } from '@angular/material/legacy-button';
import { MatLegacyCardModule as MatCardModule } from '@angular/material/legacy-card';
import { MatLegacyDialogModule as MatDialogModule } from '@angular/material/legacy-dialog';
import { MatLegacyFormFieldModule as MatFormFieldModule } from '@angular/material/legacy-form-field';
import { MatLegacyInputModule as MatInputModule } from '@angular/material/legacy-input';
import { MatLegacySelectModule as MatSelectModule } from '@angular/material/legacy-select';
import { MatLegacySlideToggleModule as MatSlideToggleModule } from '@angular/material/legacy-slide-toggle';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatStepperModule } from '@angular/material/stepper';

@NgModule({
  imports: [
    CommonModule,
    MatCardModule,
    MatSelectModule,
    MatButtonModule,
    MatFormFieldModule,
    MatInputModule,
    MatStepperModule,
    MatSidenavModule,
    MatExpansionModule,
    MatIconModule,
    MatDialogModule,
    MatSlideToggleModule,
  ],
  exports: [
    CommonModule,
    MatCardModule,
    MatSelectModule,
    MatButtonModule,
    MatFormFieldModule,
    MatInputModule,
    MatStepperModule,
    MatSidenavModule,
    MatExpansionModule,
    MatIconModule,
    MatDialogModule,
    MatSlideToggleModule,
  ],
})
export class MaterialModule {}
