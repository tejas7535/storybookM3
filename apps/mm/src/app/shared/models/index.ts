export * from './app-delivery.model';
export * from './bearing-search';
export * from './home';
export * from './lazy-list-loader';
export * from './maintenance-message';
export * from './preflight-request';
export * from './result-page';
