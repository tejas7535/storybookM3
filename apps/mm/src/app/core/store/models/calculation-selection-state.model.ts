import { BearingOption } from '@mm/shared/models';
import { ListValue } from '@mm/shared/models/lazy-list-loader/mm-list-value.model';

export interface CalculationSelectionState {
  bearingResultList: BearingOption[];
  bearing?: Bearing;
  bearingSeats?: StepSelectionValue | undefined;
  measurementMethods?: StepSelectionValue | undefined;
  mountingMethods?: StepSelectionValue | undefined;
  stepper: {
    currentStep: number;
  };
  loading?: boolean;
}

export interface Bearing {
  bearingId: string;
  title: string;
  type?: {
    typeId: string;
    title: string;
  };
  series?: {
    seriesId: string;
    title: string;
  };
}

export interface StepSelectionValue {
  values: ListValue[];
  selectedValueId?: string;
}
