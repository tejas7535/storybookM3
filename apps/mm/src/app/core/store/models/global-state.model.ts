import { AppDelivery } from '@mm/shared/models';

export interface GlobalState {
  isStandalone: boolean;
  appDelivery: `${AppDelivery}`;
  initialized: boolean;
  isInternalUser: boolean;
}
