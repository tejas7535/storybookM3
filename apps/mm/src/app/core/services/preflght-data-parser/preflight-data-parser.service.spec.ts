import { MMBearingPreflightResponse } from '@mm/shared/models';
import { PREFLIGHT_SUCESS_RESPONSE_MOCK } from '@mm/testing/mocks';
import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';

import { PreflightDataParserService } from './preflight-data-parser.service';

describe('PreflightDataParserService', () => {
  let spectator: SpectatorService<PreflightDataParserService>;
  const createService = createServiceFactory(PreflightDataParserService);

  beforeEach(() => {
    spectator = createService();
  });

  it('should be created', () => {
    expect(spectator.service).toBeTruthy();
  });

  it('should format preflight data correctly', () => {
    const mockData: MMBearingPreflightResponse = PREFLIGHT_SUCESS_RESPONSE_MOCK;

    const result = spectator.service.formatPreflightData(mockData);

    expect(result).toMatchSnapshot();
  });

  it('should throw an error if hydraulic nut type field is missing', () => {
    const mockData: MMBearingPreflightResponse = {
      ...PREFLIGHT_SUCESS_RESPONSE_MOCK,
      data: {
        input: [
          {
            fields: [
              {
                id: 'IDMM_MODULUS_OF_ELASTICITY',
                defaultValue: 'modulus1',
                range: [],
              },
            ],
          },
        ],
      },
    } as Partial<MMBearingPreflightResponse> as MMBearingPreflightResponse;

    expect(() => spectator.service.formatPreflightData(mockData)).toThrow(
      'Cannot find IDMM_HYDRAULIC_NUT_TYPE field'
    );
  });

  it('when there is missing data', () => {
    const mockData: MMBearingPreflightResponse = {
      ...PREFLIGHT_SUCESS_RESPONSE_MOCK,
      data: {
        input: undefined,
      },
    } as Partial<MMBearingPreflightResponse> as MMBearingPreflightResponse;

    expect(() => spectator.service.formatPreflightData(mockData)).toThrow(
      'Cannot find IDMM_HYDRAULIC_NUT_TYPE field'
    );
  });
});
