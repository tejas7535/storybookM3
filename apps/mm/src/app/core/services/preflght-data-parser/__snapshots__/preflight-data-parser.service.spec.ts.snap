// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreflightDataParserService should format preflight data correctly 1`] = `
Object {
  "hudraulicNutType": Object {
    "options": Array [
      Object {
        "label": "HYDNUT1000-E",
        "value": "LB_HYDNUT1000_E",
      },
    ],
    "value": "LB_HYDNUT1000_E",
  },
  "innerRingExpansion": "0.450",
  "modulusOfElasticity": "210000",
  "mountingOption": "LB_RADIAL_CLEARANCE_REDUCTION",
  "numberOfPreviousMountings": "LB_ZERO_TO_ONE",
  "poissonRatio": "0.3",
  "radialClearanceReduction": "450.0",
  "shaftDiameter": "0",
  "shaftMaterial": "STEEL_20_DEGREE",
}
`;
